### ISSUE TITLE
Nano : quantize dataloader preprocessing

### ISSUE THREAD

**Issue by rnwang04** [intel-analytics/BigDL#4924]
When I experiment with Transformer accelerating with Nano, I meet similar problems in two repos.
BST : https://github.com/jiwidi/Behavior-Sequence-Transformer-Pytorch
TS Transformer : https://github.com/thuml/Autoformer

when quantize, it will report TypeError: forward() takes 2 positional arguments but 9 were given.
In BST，the output data of train_data_loader is : `user_id, movie_history, target_movie_id,  movie_history_ratings, target_movie_rating, sex, age_group, occupation`
In TS Transfomrt，the output data of train_data_loader is : `seq_x, seq_y, seq_x_mark, seq_y_mark`
But quantize with None support only the above two formats:
`for idx, (input, label) in enumerate(dataloader):`
`for idx, input in enumerate(dataloader):`
Therefore, there should be some preprocessing for quantization.

My current workaround for TS Transformer:
```
def wrapper(collate_fn):
        def new_fn(data):
            data = collate_fn(data)
            batch_x, batch_y, batch_x_mark, batch_y_mark = map(lambda x: x.float(), data)
            dec_inp = torch.zeros_like(batch_y[:, -args.pred_len:, :]).float()
            dec_inp = torch.cat([batch_y[:, :args.label_len, :], dec_inp], dim=1).float()
            input_data = (batch_x, batch_x_mark, dec_inp, batch_y_mark)
            return input_data, batch_y
        return new_fn
# wrap preprocessing  to collate_fn
train_data_loader.collate_fn = wrapper(train_data_loader.collate_fn)
```
Then `quantize_model = Trainer.quantize(model, calib_dataloader=train_data_loader)` works. 
💡I wonder if it is possible to add a preprocess function parameter to quantize, and then wrap it into collate_fn in Nano code? Is this appropriate?
🐛Another problem is above preprocessing doesn't work for quantize with openvino.

For BST, its input of  forward function is batch which is a list, and  in forward function the conversion looks like : `transfomer_features, user_features, target_movie_rating = self.encode_input(batch)`
But in inc, it does such preprocess: 
```
isinstance(input, list) or isinstance(input, tuple):
        if device=='cpu':
            output = model(*input)
```
and BST needs just `output = model(input)`.
🐛I don't find a suitable and elegant way to solve this bug now.

**Comment by zhentaocc** [comment]
1. If your BST pipeline includes openvino/onnx acceleration,  I think you have to change the forward definition to `forward(args1, args2, args3, ...)`.
2.  I think we can provide a function or a few arguments for users to reformat the data. But this is quite difficult to cover all cases. It's simple to have a function on your autoformer issue, but uses can always have a different data format you've never seen...

**Comment by zhentaocc** [comment]
BTW, are you able to use BST with onnxruntime?

**Comment by rnwang04** [comment]
> BTW, are you able to use BST with onnxruntime?

yes, https://github.com/intel-analytics/BigDL/issues/4970 has reported the bug. I fix it by manually remove useless input and add a length check for this type of problem.

**Comment by rnwang04** [comment]
> 1. If your BST pipeline includes openvino/onnx acceleration,  I think you have to change the forward definition to `forward(args1, args2, args3, ...)`.
> 2. I think we can provide a function or a few arguments for users to reformat the data. But this is quite difficult to cover all cases. It's simple to have a function on your autoformer issue, but uses can always have a different data format you've never seen...

1. yeah, I have modified the forward definition in BST.
2. I agree. It's difficult to cover all formats. But maybe we can cover part of them?
